server/text_generation_server/utils/gptq/exllamav2.py

```diff
@@ -185,6 +188,11 @@ def post_init(self, temp_dq):
             "g_idx": self.g_idx,
         }
         temp_dq = temp_dq.get_scratch_slice(self.temp_dq_size())
+
+        # We NEED to keep a pointer on Python side, otherwise the garbage collector will mess with us,
+        # and `Memory access fault by GPU node-2` will EAT you.
+        self.temp_dq = temp_dq

```

Comment:
This gave me headaches today.

Reply:
THIS is what was crashing TP>1?

Reply:
@narsil can we test this quickly? If that's the case that would be amazing.

Reply:
https://github.com/huggingface/text-generation-inference/pull/1490